### PR TITLE
feat(superuser): Add badge and warning alert

### PR DESCRIPTION
**Pull request by leedongwei**
Hidden on self-hosted and "demo" organization

Screenshot 2024-01-02 at 2 28 03 PM

### CODE REVIEW

File: static/app/components/sidebar/index.tsx [hunk 1]
```diff
@@ -118,7 +119,9 @@ function Sidebar({organization}: Props) {
 
   const collapsed = !!preferences.collapsed;
   const horizontal = useMedia(`(max-width: ${theme.breakpoints.medium})`);
-  const hasSuperuserSession = isActiveSuperuser();
+
+  // Avoid showing superuser UI on self-hosted instances
+  const hasSuperuserSession = isActiveSuperuser() && !ConfigStore.get('isSelfHosted');
```
Comment (ryan953) [inline static/app/components/sidebar/index.tsx:124]: I was surprised to see `isActiveSuperuser` in sentry repo and not getsentry.

**Review by ryan953 — APPROVED** [review]

File: static/app/components/sidebar/index.tsx (hunk 1 above)
Comment (markstory) [reply static/app/components/sidebar/index.tsx:124]: self-hosted also uses superuser for instance level administration views and endpoints.

**Review by markstory — APPROVED** [review]

File: static/app/components/sidebar/index.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/components/sidebar/index.tsx:124]: oh

@leedongwei should the warning/alert be a hook? We could have getsentry implement the "you'll get fired" part, and for self-hosted and other contexts the alert still be red across the top and say 'You are in superuser mode.'

Comment (leedongwei) [reply static/app/components/sidebar/index.tsx:124]: @ryan953 Yea it's been moved into a hook.

**Pull request merged**